firmware/src/drivers/spi.h

```diff
@@ -18,27 +18,23 @@
 
 #pragma once
 
-#include "cmsis_os.h"
 #include "target.h"
 
-enum cs_type_e {
-  LOW_ACTIVE = GPIO_PIN_RESET,
-  HIGH_ACTIVE = GPIO_PIN_SET,
-};
+namespace drivers {
 
-struct SPI_BUS {
-  GPIO_TypeDef* const cs_port;
-  uint16_t cs_pin;
-  cs_type_e cs_type;
-  SPI_HandleTypeDef* const spi_handle;
-  uint8_t initialized;
-  bool busy;
-};
+class Spi {
+ public:
+  explicit Spi(SPI_HandleTypeDef* spi_handle, uint32_t timeout = 5U) : m_spi_handle(spi_handle), M_TIMEOUT(timeout) {}
+
+  void Transfer(uint8_t* data, const size_t length) { HAL_SPI_Transmit(m_spi_handle, data, length, M_TIMEOUT); }
```

Comment:
can it be `const uint8_t* data`?

Reply:
It can't, for some reason the HAL function wants a non constant array...